moxin-protocol/src/protocol.rs

```diff
@@ -1,4 +1,42 @@
-use crate::data::Model;
+use crate::data::*;
+use chrono::NaiveDate;
+
+#[derive(Clone, Debug)]
+pub enum ContextOverflowPolicy {
+    StopAtLimit,
+    TruncateMiddle,
+    TruncatePastMessages,
+}
+
+#[derive(Clone, Debug)]
+pub enum GPULayers {
+    Specific(u32),
+    Max,
+}
+
+#[derive(Clone, Debug)]
+pub struct ModelExecutionConfig {
+    pub cpu_threads: u32,
+    pub gpu_layers: GPULayers,
+    pub use_mlock: bool,
+    pub n_batch: u32,
+    pub n_ctx: u32,
+    pub rope_freq_scale: f32,
+    pub rope_freq_base: f32,
+
+    // TBD Not really sure if this is something backend manages or if it is matter of
+    // the client (if it is done by tweaking the JSON payload for the chat completition)
+    pub context_overflow_policy: ContextOverflowPolicy
+}
```

Comment:
I would like you to take a look at `ModelExecutionConfig`, referencing [`LoadModel`](https://github.com/project-robius/moxin/pull/2/files#diff-d2782c1a1438e2421fa81b5c49fd795ff38e3c4f85caaa4e60a5490303edc397R27-R44), where `reverse_prompt` and `prompt_template` are important parameters for some models.

Reply:
I got it. I think it is a valid alternative, as long it makes sense on your end. I will change it 👍

Reply:
BTW Looking again at LM Studio, I found that `cpu_threads` is something that can change with each chat interaction, so if you're ok I'm gonna remove from this list and include it as part of the JSON payload for each chat message (where other related values such us `temp`, `n_predict`, goes)

<img width="315" alt="image" src="https://github.com/project-robius/moxin/assets/487140/ab5a2f82-007a-4e2e-9024-28dd5a41f3ef">

Reply:
https://github.com/WasmEdge/WasmEdge/blob/7c13bee99c796317fc68d5bf46d737f9b51d06fe/plugins/wasi_nn/ggml.cpp#L52-L57
We have these parameters, but we didn't see min-p and top-k. Are these required?

Reply:
These changes are now implemented